docs/extensions/storage/proposedchanges.md

```diff
@@ -0,0 +1,185 @@
+- iterating over the root of the storage and treating every folder following a new `<projectlayout>` as a `project` *storage space*, or
+- iterating over the root of the storage and treating every folder following a generic `<layout>` as a *storage space* for a configurable space type, or
+- we allow configuring a map of `space type` to `layout` (based on the [CreateStorageSpaceRequest](https://cs3org.github.io/cs3apis/#cs3.storage.provider.v1beta1.CreateStorageSpaceRequest)) which would allow things like
+```
+home=/var/lib/ocis/storage/home/{{substr 0 1 .Owner.Username}}/{{.Owner.Username}}
+spaces=/spaces/var/lib/ocis/storage/projects/{{.Name}}
+```
+
+This would make the `GetHome()` call return the path to the *storage provider* including the relative path to the *storage space*. No need for a *storage provider* mounted at `/home`. This is just a UI alias for `/users/<userlayout>`. Just like a normal `/home/<username>` on a linux machine.
+
+But if we have no `/home` where do we find the shares, and how can clients discover changes in accepted shares?
+
+The `/shares` namespace should be provided by a *shares storage provider* that lists all accepted shares for the current user... but what about copy pasting links from the browser? Well this storage is only really needed to have a path to ocm shares that actually reside on other instances. In the UI the shares would be listed by querying a *share manager*. It returns ResourceIds, which can be stated to fetch a path that is then accessible in the CS3 global namespace. Two caveats:
+- This only works for resources that are actually hosted by the current instance. For those it would leak the parent path segments to a shared resource.
+- For accepted OCM shares there must be a path in the [*CS3 global namespace*]({{< ref "./namespaces.md#cs3-global-namespaces" >}}) that has to be the same for all users, otherwise they cannot copy and share those URLs.
+
+Work on this is done in https://github.com/cs3org/reva/pull/1846
+
+### The gateway should be responsible for path transformations
+
+Currently, storage providers are aware af their mount point, coupling them tightly with the gateway.
+
+Tracked in https://github.com/cs3org/reva/issues/578
+
+Work is done in https://github.com/cs3org/reva/pull/1866
+
+## URL escaped string representation of a CS3 reference
+
+For the `/dav/spaces/` endpoint we need to encode the *reference* in a url compatible way. 
+1. We can separate the path using a `/`: `/dav/spaces/<spaceid>/<path>`
+2. The `spaceid` currently is a cs3 resourceid, consisting of `<storageid>` and `<nodeid>`. Since the nodeid might contain `/` eg. for the local driver we have to urlencode the spaceid.
+
+To access resources by id we need to make the `/dav/meta/<resourceid>` able to list directories... Otherwise id based navigation first has to look up the path. Or we use the libregraph api for id based navigation.
```

Comment:
Same for `/dev/meta` - where are these two endpoints coming from? Part of REVA?

Reply:
`/dav/meta` already exists in oc10 and is the only api endpoint that allows looking up a file by id. I'll clarify that